File: mcp-code-intelligence-kotlin/src/main/kotlin/com/codeintel/context/IntentStrategies.kt
/** Intent Strategies — maps intent to prioritized section list. KSA-171. */
package com.codeintel.context

data class SectionDef(val name: String, val priority: Int, val format: String)

data class IntentStrategy(val intent: String, val sections: List<SectionDef>)

private val STRATEGIES = mapOf(
    "explain" to IntentStrategy("explain", listOf(
        SectionDef("source", 1, "full"),
        SectionDef("doc_comment", 2, "full"),
        SectionDef("siblings", 3, "signatures"),
        SectionDef("imports", 4, "full"),
        SectionDef("callers", 5, "summary"),
        SectionDef("callees", 6, "summary"),
        SectionDef("type_definitions", 7, "full"),
    )),
    "modify" to IntentStrategy("modify", listOf(
        SectionDef("source", 1, "full"),
        SectionDef("callers", 2, "full"),
        SectionDef("callees", 3, "full"),
        SectionDef("tests", 4, "full"),
        SectionDef("imports", 5, "full"),
        SectionDef("type_definitions", 6, "full"),
        SectionDef("siblings", 7, "signatures"),
    )),
    "debug" to IntentStrategy("debug", listOf(
        SectionDef("source", 1, "full"),
        SectionDef("callers", 2, "full"),
        SectionDef("error_patterns", 3, "full"),
        SectionDef("recent_changes", 4, "full"),
        SectionDef("imports", 5, "full"),
        SectionDef("siblings", 6, "signatures"),
        SectionDef("callees", 7, "summary"),
    )),
    "test" to IntentStrategy("test", listOf(
        SectionDef("source", 1, "full"),
        SectionDef("tests", 2, "full"),
        SectionDef("test_patterns", 3, "full"),
        SectionDef("callees", 4, "full"),
        SectionDef("type_definitions", 5, "full"),
        SectionDef("mocks_needed", 6, "full"),
        SectionDef("siblings", 7, "signatures"),
    )),
)

/** Get the intent strategy for a given intent. Falls back to 'explain'. */
fun getStrategy(intent: String): IntentStrategy =
    STRATEGIES[intent] ?: STRATEGIES["explain"]!!

/** Get all supported intent names. */
fun getSupportedIntents(): List<String> = STRATEGIES.keys.toList()

File: backend/mcp-code-intelligence-nodejs/src/context/intent-strategies.ts
/**
 * KSA-158: Intent Strategies — maps intent to prioritized section list.
 */

export interface SectionDef {
  name: string;
  priority: number;
  format: 'full' | 'summary' | 'signatures';
}

export interface IntentStrategy {
  intent: string;
  sections: SectionDef[];
}

const STRATEGIES: Record<string, IntentStrategy> = {
  explain: {
    intent: 'explain',
    sections: [
      { name: 'source', priority: 1, format: 'full' },
      { name: 'doc_comment', priority: 2, format: 'full' },
      { name: 'siblings', priority: 3, format: 'signatures' },
      { name: 'imports', priority: 4, format: 'full' },
      { name: 'callers', priority: 5, format: 'summary' },
      { name: 'callees', priority: 6, format: 'summary' },
      { name: 'type_definitions', priority: 7, format: 'full' },
    ]
  },
  modify: {
    intent: 'modify',
    sections: [
      { name: 'source', priority: 1, format: 'full' },
      { name: 'callers', priority: 2, format: 'full' },
      { name: 'callees', priority: 3, format: 'full' },
      { name: 'tests', priority: 4, format: 'full' },
      { name: 'imports', priority: 5, format: 'full' },
      { name: 'type_definitions', priority: 6, format: 'full' },
      { name: 'siblings', priority: 7, format: 'signatures' },
    ]
  },
  debug: {
    intent: 'debug',
    sections: [
      { name: 'source', priority: 1, format: 'full' },
      { name: 'callers', priority: 2, format: 'full' },
      { name: 'error_patterns', priority: 3, format: 'full' },
      { name: 'recent_changes', priority: 4, format: 'full' },
      { name: 'imports', priority: 5, format: 'full' },
      { name: 'siblings', priority: 6, format: 'signatures' },
      { name: 'callees', priority: 7, format: 'summary' },
    ]
  },
  test: {
    intent: 'test',
    sections: [
      { name: 'source', priority: 1, format: 'full' },
      { name: 'tests', priority: 2, format: 'full' },
      { name: 'test_patterns', priority: 3, format: 'full' },
      { name: 'callees', priority: 4, format: 'full' },
      { name: 'type_definitions', priority: 5, format: 'full' },
      { name: 'mocks_needed', priority: 6, format: 'full' },
      { name: 'siblings', priority: 7, format: 'signatures' },
    ]
  }
};

/** Get the intent strategy (section priorities) for a given intent. Falls back to 'explain'. */
export function getStrategy(intent: string): IntentStrategy {
  return STRATEGIES[intent] || STRATEGIES.explain;
}

/** Get all supported intent names. */
export function getSupportedIntents(): string[] {
  return Object.keys(STRATEGIES);
}
